### PR TITLE
Second week exercise Exercise four 

### PR DESCRIPTION
**Pull request by fluty84**
I don't know if I need to return the credit in the las 500 error of sendMessage... 
` postReq.on("error", (error) => {
    res.statusCode = 500;
    res.end(error.message);
  });`

### CODE REVIEW

File: src/controllers/sendMessage.js [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
-  const body = JSON.stringify(req.body);
+
+  const body = JSON.stringify(req.body)
+
+  const actualMoney = await checkBudget()
+
+  if (actualMoney.amount < MESSAGE_PRICE) {
+    return res.status(500).json("Please Top Up")
+  } else {
+    saveAmount(-MESSAGE_PRICE)
```
Comment (Teo-Lopez) [inline src/controllers/sendMessage.js:19]: Be careful with the naming, saveAmount doesn't save a new amount, it changes the credit by X amount. A better name could be changeBudgetBy(-2)

File: src/clients/saveMessage.js [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+      console.log("Message saved succesfully:", message);
+      mutex.unlock();
+    });
 
-    console.log("Message saved succesfully:", doc);
-    return doc;
+    return message;
   } catch (err) {
+    mutex.unlock();
     console.log("Error while saving", err);
```
Comment (Teo-Lopez) [inline src/clients/saveMessage.js:24]: You can add a .finally(()=> mutex.unlock())

**Review by Teo-Lopez — COMMENTED** [review]
About returning the amount on the 500 error is just a decision on how you want it to behave, as long as you save back the correct amount and inform the user that he wasn't charged giving more details is optional

**Comment by fluty84** [comment]
thanx!

**Pull request merged**